tests/foreman/ui/test_acs.py

```diff
@@ -0,0 +1,357 @@
+"""Tests for Alternate Content Sources UI
+
+:Requirement: AlternateContentSources
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: AlternateContentSources
+
+:Team: Phoenix-content
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import pytest
+
+from robottelo import constants
+from robottelo.config import settings
+from robottelo.constants import CONTENT_CREDENTIALS_TYPES
+from robottelo.constants import REPO_TYPE
+from robottelo.utils.datafactory import gen_string
+
+
+ssl_name, product_name, product_label, product_description = (gen_string('alpha') for _ in range(4))
+repos_to_enable = ['rhae2.9_el8']
+
+
+@pytest.fixture(scope='function')
+def ui_acs_setup(session, target_sat, module_sca_manifest_org):
+    """
+    This fixture creates all the necessary data for the test to run.
+    It creates an organization, content credentials, product and repositories.
+    """
+
+    with target_sat.ui_session() as session:
+        session.organization.select(org_name=module_sca_manifest_org.name)
+
+        session.contentcredential.create(
+            {
+                'name': ssl_name,
+                'content_type': CONTENT_CREDENTIALS_TYPES['ssl'],
+                'content': gen_string('alpha'),
+            }
+        )
+
+        session.product.create(
+            {'name': product_name, 'label': product_label, 'description': product_description}
+        )
+
+        session.repository.create(
+            product_name,
+            {
+                'name': gen_string('alpha'),
+                'repo_type': REPO_TYPE['file'],
+                'repo_content.upstream_url': settings.repos.file_type_repo.url,
+            },
+        )
+
+        for repo in repos_to_enable:
+            session.redhatrepository.enable(
+                custom_query=constants.REPOS[repo]['id'],
+                arch=constants.DEFAULT_ARCHITECTURE,
+                entity_name=None,
+            )
+
+        return target_sat, module_sca_manifest_org
+
+
+@pytest.mark.e2e
+def test_acs_positive_end_to_end(session, ui_acs_setup):
+    """
+    Create, update, delete and refresh ACSes of all supported types.
+
+    :id: 047452cc-5a9f-4473-96b1-d5b6830b7d6b
+
+    :steps:
+        1. Select an organization
+        2. Create ACSes (all supported types/combinations)
+        3. Create ACS on which deletion is going to be tested
+        4. Test deletion
+        5. Test refresh
+        6. Test renaming and changing description
+        7. Test editing capsules
+        8. Test editing urls and subpaths
+        9. Test editing credentials
+        10. Test editing products
+
```

Comment:
Codewise I see no problems but I'm not convinced of this all being in a single test. If there is a bug in creation of just of the types in step 2 we never get to testing other things, with 10 types it is not so improbable.

IIUC the ACSes in edit and renamed steps are just selected randomly, and there are special ones created to test deletion anyway. 

My suggestion is to have a separate test for step 2 -- `check all types can be created`. 

And then have a separate `e2e test` for the other steps, just move the deletion part at the very end. 

Bonus suggestion: consider what it would take to have ACS configuration for the individual types defined in some structure and then use that to parametrize the tests. For the `check all types can be created` test it would ensure all types are always tested (currently, if the first one fails others are not visited). For the `e2e` test, we could pick one or more types for coverage, or a random one perhaps :)